packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.types.ts

```diff
@@ -0,0 +1,67 @@
+import { TreeViewPluginSignature } from '../../models';
+import { TreeViewItemId } from '../../../models';
+import { UseTreeViewItemsSignature } from '../useTreeViewItems';
+
+export interface UseTreeViewLabelPublicAPI {
+  /**
+   * Used to update the label of an item.
+   * @param {TreeViewItemId} itemId The id of the item that is being edited.
+   * @param {string} newLabel The new label of the item.
+   */
+  updateItemLabel: (itemId: TreeViewItemId, newLabel: string) => void;
+}
+
+export interface UseTreeViewLabelInstance extends UseTreeViewLabelPublicAPI {
+  /**
+   * Used to set the state.
+   * @param {TreeViewItemId} itemId The item that is being currently edited.
+   * @returns {void}.
+   */
+  setEditedItemId: (itemId: TreeViewItemId | null) => void;
+  /**
+   * Used to determine if an item is currently being edited.
+   * @param {TreeViewItemId} itemId The item to check.
+   * @returns {void}.
+   */
+  isItemBeingEdited: (itemId: TreeViewItemId) => boolean;
+  /**
+   * Used to determine if a given item is editable.
+   * @param {TreeViewItemId} itemId The item to check.
+   * @returns {boolean} `true` if the item should be editable.
+   */
+  isItemEditable: (itemId: TreeViewItemId) => boolean;
+  /**
+   * Used to determine if the tree view is editable.
```

Comment:
I propose to be extra explicit that it's a boolean here, since then `instance` usually hosts methods:

```suggestion
   * Set to `true` if the tree view is editable.
```